### ISSUE TITLE
Container retained in mqtt after deletion 

### ISSUE THREAD

**Issue by Isa-78** [getwud/wud#216]
Hi @fmartinou , 
I just noticed that when deleting a container from a docker, it doesn't remove the information from the mqtt server. 
Which means that on each cron, I receive an update message in mqtt for a container that doesn't exist anymore. 
Can you please check if you can pass the info to mqtt and remove the container? 
Or is there something that I've missed in the config? 
Many thanks,
Isa

**Comment by the-jeffski** [comment]
Yep, have been seeing the same so I have to go to Mosquitto and clear out the entries.

**Comment by geekifier** [comment]
I'd like to add, this also happens if a previously watched container has the `wud.watch=false` label applied. It disappears from the WUD UI, but persists its state in Home Assistant.

**Comment by fmartinou** [comment]
Hi @Isa-78,

Thanks for reaching out!

This is how MQTT retained messages operate: WUD publishes Home Assistant MQTT discovery topics with the `retained` flag enabled so that Home Assistant immediately discovers entities upon reboot.

When a container is deleted from Docker, the MQTT broker retains the last published message indefinitely until it is explicitly cleared. 

To remove deleted container entities from Home Assistant:
- Publish an empty (null) retained payload to the container's discovery topic on your MQTT broker.
- Or restart the broker / clean up retained topics.

Closing as standard MQTT retained message behavior.
